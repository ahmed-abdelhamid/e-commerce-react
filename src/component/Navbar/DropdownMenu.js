import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../action/auth/auth';

export const DropdownMenu = (props) => (
  <div className="navbar-nav nav-item dropdown">
    <span className="nav-link dropdown-toggle" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
      {props.fullname}
    </span>
    <div className="dropdown-menu" aria-labelledby="navbarDropdown">
      <Link to="/edit" className="dropdown-item">Edit Profile</Link>
      <Link to="/setting" className="dropdown-item">Setting</Link>
      <button
        className="dropdown-item"
        onClick={() => props.logout()}
      >
        Logout
      </button>
    </div>
  </div>
)

const mapStateToProps = ({ auth }) => ({
  fullname: `${auth.firstName} ${auth.lastName}`
});

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(DropdownMenu);