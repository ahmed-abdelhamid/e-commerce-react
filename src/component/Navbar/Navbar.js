import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const Navbar = () => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
    <div className="container">
      <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
        <span className="navbar-toggler-icon"></span>
      </button>

      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <div className="navbar-nav mr-auto">
          <NavLink to="/dashboard" className="nav-item nav-link" activeClassName="active">Home</NavLink>
          <NavLink to="/categories" className="nav-item nav-link" activeClassName="active">Categories</NavLink>
          <NavLink to="items" className="nav-item nav-link" activeClassName="active">Items</NavLink>
          <NavLink to="/members" className="nav-item nav-link" activeClassName="active">Members</NavLink>
          <NavLink to="/statistics" className="nav-item nav-link" activeClassName="active">Statistics</NavLink>
          <NavLink to="/logs" className="nav-item nav-link" activeClassName="active">Logs</NavLink>
        </div>
      </div>
      <div className="navbar-nav nav-item dropdown">
        <span className="nav-link dropdown-toggle" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
          $Username
        </span>
        <div className="dropdown-menu" aria-labelledby="navbarDropdown">
          <Link to="/edit" className="dropdown-item">Edit Profile</Link>
          <Link to="/setting" className="dropdown-item">Setting</Link>
          <Link to="/" className="dropdown-item">Logout</Link>
        </div>
      </div>
    </div>
  </nav>
);

export default Navbar;