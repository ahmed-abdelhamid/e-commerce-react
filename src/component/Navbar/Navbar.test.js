import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Navbar from './Navbar';

test('shoould render navbar', () => {
  const wrapper = shallow(<Navbar />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});