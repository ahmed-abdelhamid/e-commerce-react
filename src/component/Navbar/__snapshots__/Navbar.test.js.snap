// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoould render navbar 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <div
        className="navbar-nav mr-auto"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/dashboard"
        >
          Home
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/categories"
        >
          Categories
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="items"
        >
          Items
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/members"
        >
          Members
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/statistics"
        >
          Statistics
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/logs"
        >
          Logs
        </NavLink>
      </div>
    </div>
    <Connect(DropdownMenu) />
  </div>
</nav>
`;
